Module for Magento XXE  (CVE-2024-34102) (#19304)
rapid7/metasploit-framework: documentation/modules/auxiliary/gather/magento_xxe_cve_2024_34102.md

    volumes:
      - 'mariadb_data:/bitnami/mariadb'
  magento:
    image: docker.io/bitnami/magento:2
    ports:
      - '80:8080'
      - '443:8443'
    environment:
      - MAGENTO_HOST=localhost
      - MAGENTO_DATABASE_HOST=mariadb
      - MAGENTO_DATABASE_PORT_NUMBER=3306
      - MAGENTO_DATABASE_USER=bn_magento
      - MAGENTO_DATABASE_NAME=bitnami_magento
      - ELASTICSEARCH_HOST=elasticsearch
      - ELASTICSEARCH_PORT_NUMBER=9200
      # ALLOW_EMPTY_PASSWORD is recommended only for development.
      - ALLOW_EMPTY_PASSWORD=yes
    volumes:
      - 'magento_data:/bitnami/magento'
    depends_on:
      - mariadb
      - elasticsearch
  elasticsearch:
    image: docker.io/bitnami/elasticsearch:7
    volumes:
      - 'elasticsearch_data:/bitnami/elasticsearch/data'
volumes:
  mariadb_data:
    driver: local
  magento_data:
    driver: local
  elasticsearch_data:
    driver: local
```

Run the below command to create the container:

```
$ docker-compose up
```

## Verification Steps
Follow [Setup](#setup) and [Scenarios](#scenarios).

## Options

### TARGETURI (required)

The path to the Magento (Default: `/`).

### SRVHOST (required)

> The local host or network interface to listen on. This must be an address on the local machine or 0.0.0.0 to listen on all addresses.

Reviewer: ```suggestion
The local IP address to listen on. This must be a routable IP address on the local machine (0.0.0.0 is invalid). 
```
